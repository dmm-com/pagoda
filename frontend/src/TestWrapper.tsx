import { createTheme, ThemeProvider } from "@mui/material/styles";
import { SnackbarProvider } from "notistack";
import React, { FC } from "react";
import { MemoryRouter } from "react-router";

export const TestWrapper: FC<{ children: React.ReactNode }> = ({
  children,
}) => {
  const theme = createTheme();
  return (
    <ThemeProvider theme={theme}>
      <SnackbarProvider maxSnack={1} autoHideDuration={100}>
        <MemoryRouter>{children}</MemoryRouter>
      </SnackbarProvider>
    </ThemeProvider>
  );
};

export const TestWrapperWithoutRoutes: FC<{ children: React.ReactNode }> = ({
  children,
}) => {
  const theme = createTheme();
  return (
    <ThemeProvider theme={theme}>
      <SnackbarProvider maxSnack={1} autoHideDuration={100}>
        {children}
      </SnackbarProvider>
    </ThemeProvider>
  );
};
