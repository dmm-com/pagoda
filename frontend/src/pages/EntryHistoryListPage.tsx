import AppsIcon from "@mui/icons-material/Apps";
import { Box, Container, Divider, IconButton, Typography } from "@mui/material";
import { FC, useState } from "react";

import { useAsyncWithThrow } from "../hooks/useAsyncWithThrow";

import { Loading } from "components/common/Loading";
import { PageHeader } from "components/common/PageHeader";
import { EntryBreadcrumbs } from "components/entry/EntryBreadcrumbs";
import { EntryControlMenu } from "components/entry/EntryControlMenu";
import { EntryHistoryList } from "components/entry/EntryHistoryList";
import { EntrySelfHistoryList } from "components/entry/EntrySelfHistoryList";
import { usePage } from "hooks/usePage";
import { useTypedParams } from "hooks/useTypedParams";
import { aironeApiClient } from "repository/AironeApiClient";

export const EntryHistoryListPage: FC = () => {
  const { entryId } = useTypedParams<{ entityId: number; entryId: number }>();

  const { page: attributeHistoryPage, changePage: changeAttributeHistoryPage } =
    usePage();
  const { page: selfHistoryPage, changePage: changeSelfHistoryPage } =
    usePage();

  const [entryAnchorEl, setEntryAnchorEl] = useState<HTMLButtonElement | null>(
    null,
  );

  const entry = useAsyncWithThrow(async () => {
    return entryId != undefined
      ? await aironeApiClient.getEntry(entryId)
      : undefined;
  }, [entryId]);

  const histories = useAsyncWithThrow(async () => {
    return await aironeApiClient.getEntryHistories(
      entryId,
      attributeHistoryPage,
    );
  }, [entryId, attributeHistoryPage]);

  const selfHistories = useAsyncWithThrow(async () => {
    return await aironeApiClient.getEntrySelfHistories(
      entryId,
      selfHistoryPage,
    );
  }, [entryId, selfHistoryPage]);

  return (
    <Box className="container">
      <EntryBreadcrumbs entry={entry.value} title="変更履歴" />

      <PageHeader title={entry.value?.name ?? ""} description="変更履歴">
        <Box width="50px">
          <IconButton
            onClick={(e) => {
              setEntryAnchorEl(e.currentTarget);
            }}
          >
            <AppsIcon />
          </IconButton>
          <EntryControlMenu
            entityId={entry.value?.schema?.id ?? 0}
            entryId={entryId}
            anchorElem={entryAnchorEl}
            handleClose={() => setEntryAnchorEl(null)}
          />
        </Box>
      </PageHeader>

      <Container>
        <Box sx={{ mb: 4 }}>
          <Typography
            variant="h6"
            sx={{ mb: 2, color: "#455A64", fontWeight: "bold" }}
          >
            アイテム変更履歴
          </Typography>
          {selfHistories.loading || !selfHistories.value ? (
            <Loading />
          ) : (
            <EntrySelfHistoryList
              entityId={entry.value?.schema?.id ?? 0}
              entryId={entryId}
              histories={selfHistories.value}
              page={selfHistoryPage}
              changePage={changeSelfHistoryPage}
            />
          )}
        </Box>

        <Divider sx={{ my: 3 }} />

        <Box>
          <Typography
            variant="h6"
            sx={{ mb: 2, color: "#455A64", fontWeight: "bold" }}
          >
            属性変更履歴
          </Typography>
          {histories.loading || !histories.value ? (
            <Loading />
          ) : (
            <EntryHistoryList
              entityId={entry.value?.schema?.id ?? 0}
              entryId={entryId}
              histories={histories.value}
              page={attributeHistoryPage}
              changePage={changeAttributeHistoryPage}
            />
          )}
        </Box>
      </Container>
    </Box>
  );
};
