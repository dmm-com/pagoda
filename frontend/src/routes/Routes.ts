export const basePath = "/ui/";

export const loginPath = () => "/auth/login/";
export const topPath = () => basePath;
export const advancedSearchPath = () => basePath + "advanced_search";
export const advancedSearchResultPath = () =>
  basePath + "advanced_search_result";
export const jobsPath = (targetId?: number) =>
  basePath + "jobs" + (targetId ? `?target_id=${targetId}` : "");
export const aclPath = (objectId: number | string) =>
  basePath + `acl/${objectId}`;
export const aclHistoryPath = (objectId: number | string) =>
  basePath + `acl/${objectId}/history`;
export const searchPath = () => basePath + "search";

// entries
export const newEntryPath = (entityId: number | string) =>
  basePath + `entities/${entityId}/entries/new`;
export const copyEntryPath = (
  entityId: number | string,
  entryId: number | string
) => basePath + `entities/${entityId}/entries/${entryId}/copy`;
export const entryDetailsPath = (
  entityId: number | string,
  entryId: number | string
) => basePath + `entities/${entityId}/entries/${entryId}/details`;
export const entryEditPath = (
  entityId: number | string,
  entryId: number | string
) => basePath + `entities/${entityId}/entries/${entryId}/edit`;
export const entityEntriesPath = (entityId: number | string) =>
  basePath + `entities/${entityId}/entries`;
export const restoreEntryPath = (entityId: number | string, keyword?: string) =>
  basePath +
  `entities/${entityId}/restore${keyword != null ? "?query=" + keyword : ""}`;
export const showEntryHistoryPath = (
  entityId: number | string,
  entryId: number | string
) => basePath + `entities/${entityId}/entries/${entryId}/history`;

// entities
export const entityHistoryPath = (entityId: number | string) =>
  basePath + `entities/${entityId}/history`;
export const listAliasPath = (entityId: number | string) =>
  basePath + `entities/${entityId}/alias`;
export const newEntityPath = () => basePath + "entities/new";
export const editEntityPath = (entityId: number | string) =>
  basePath + `entities/${entityId}`;
export const entitiesPath = () => basePath + "entities";

// triggers
export const newTriggerPath = () => basePath + "triggers/new";
export const editTriggerPath = (triggerId: number | string) =>
  basePath + `triggers/${triggerId}`;
export const triggersPath = () => basePath + "triggers";

// groups
export const newGroupPath = () => basePath + "groups/new";
export const groupPath = (groupId: number | string) =>
  basePath + `groups/${groupId}`;
export const groupsPath = () => basePath + "groups";

// users
export const newUserPath = () => basePath + "users/new";
export const userPath = (userId: number | string) =>
  basePath + `users/${userId}`;
export const usersPath = () => basePath + "users";

// roles
export const newRolePath = () => basePath + "roles/new";
export const rolePath = (roleId: number | string) =>
  basePath + `roles/${roleId}`;
export const rolesPath = () => basePath + "roles";


// category
export const newCategoryPath = () => basePath + "categories/new";
export const listCategoryPath = () => basePath + "categories/list";
export const editCategoryPath = (categoryId: number | string) => basePath + "categories/edit";