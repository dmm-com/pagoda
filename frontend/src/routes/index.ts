export * from "./AppRouter";
export * from "./Routes";
