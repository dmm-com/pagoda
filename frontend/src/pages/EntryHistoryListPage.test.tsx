/**
 * @jest-environment jsdom
 */

import { render, screen, act, waitFor } from "@testing-library/react";
import { http, HttpResponse } from "msw";
import { setupServer } from "msw/node";
import { createMemoryRouter, RouterProvider } from "react-router";

import { showEntryHistoryPath } from "../routes/Routes";

import { EntryHistoryListPage } from "./EntryHistoryListPage";

import { TestWrapperWithoutRoutes } from "TestWrapper";

const server = setupServer(
  // getEntry
  http.get("/entry/api/v2/:entryId/", () => {
    return HttpResponse.json({
      id: 1,
      name: "test entry",
      is_active: true,
      schema: {
        id: 2,
        name: "test entity",
      },
      attrs: [],
    });
  }),
  // getEntryHistories
  http.get("/entry/api/v2/:entryId/histories", () => {
    return HttpResponse.json({
      count: 0,
      results: [],
    });
  }),
  // getEntrySelfHistories
  http.get("/entry/api/v2/:entryId/self_histories/", () => {
    return HttpResponse.json({
      count: 0,
      results: [],
    });
  }),
);

beforeAll(() => server.listen());
afterEach(() => server.resetHandlers());
afterAll(() => server.close());

test("should match snapshot", async () => {
  const router = createMemoryRouter(
    [
      {
        path: showEntryHistoryPath(":entityId", ":entryId"),
        element: <EntryHistoryListPage />,
      },
    ],
    {
      initialEntries: [showEntryHistoryPath(2, 1)],
    },
  );
  const result = await act(async () => {
    return render(<RouterProvider router={router} />, {
      wrapper: TestWrapperWithoutRoutes,
    });
  });
  await waitFor(() => {
    expect(screen.queryByTestId("loading")).not.toBeInTheDocument();
  });

  expect(result).toMatchSnapshot();

  jest.clearAllMocks();
});
