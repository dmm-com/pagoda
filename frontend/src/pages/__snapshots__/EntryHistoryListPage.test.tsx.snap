// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container MuiBox-root css-0"
      >
        <div
          class="MuiBox-root css-1baykhj"
        >
          <div
            class="MuiBox-root css-t0pdf1"
          >
            <div
              class="MuiBox-root css-jqffe8"
            >
              <nav
                aria-label="breadcrumb"
                class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-5cvx94-MuiTypography-root-MuiBreadcrumbs-root"
              >
                <ol
                  class="MuiBreadcrumbs-ol css-1uwp4ue-MuiBreadcrumbs-ol"
                >
                  <li
                    class="MuiBreadcrumbs-li"
                  >
                    <a
<<<<<<< HEAD
                      class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
=======
                      class="MuiTypography-root MuiTypography-body1 css-z7oskh-MuiTypography-root"
>>>>>>> master
                      data-discover="true"
                      href="/ui/"
                    >
                      Top
                    </a>
                  </li>
                  <li
                    aria-hidden="true"
                    class="MuiBreadcrumbs-separator css-jhr02i-MuiBreadcrumbs-separator"
                  >
                    /
                  </li>
                  <li
                    class="MuiBreadcrumbs-li"
                  >
                    <a
<<<<<<< HEAD
                      class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
=======
                      class="MuiTypography-root MuiTypography-body1 css-z7oskh-MuiTypography-root"
>>>>>>> master
                      data-discover="true"
                      href="/ui/entities"
                    >
                      モデル一覧
                    </a>
                  </li>
                  <li
                    aria-hidden="true"
                    class="MuiBreadcrumbs-separator css-jhr02i-MuiBreadcrumbs-separator"
                  >
                    /
                  </li>
                  <li
                    class="MuiBreadcrumbs-li"
                  >
                    <div
                      class="MuiBox-root css-162tvoi"
                    >
                      <a
<<<<<<< HEAD
                        class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
=======
                        class="MuiTypography-root MuiTypography-body1 css-z7oskh-MuiTypography-root"
>>>>>>> master
                        data-discover="true"
                        href="/ui/entities/2/entries"
                      >
                        test entity
                      </a>
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                        data-testid="LockIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2m-6 9c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2m3.1-9H8.9V6c0-1.71 1.39-3.1 3.1-3.1s3.1 1.39 3.1 3.1z"
                        />
                      </svg>
                    </div>
                  </li>
                  <li
                    aria-hidden="true"
                    class="MuiBreadcrumbs-separator css-jhr02i-MuiBreadcrumbs-separator"
                  >
                    /
                  </li>
                  <li
                    class="MuiBreadcrumbs-li"
                  >
                    <div
                      class="MuiBox-root css-162tvoi"
                    >
                      <a
<<<<<<< HEAD
                        class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
=======
                        class="MuiTypography-root MuiTypography-body1 css-z7oskh-MuiTypography-root"
>>>>>>> master
                        data-discover="true"
                        href="/ui/entities/2/entries/1/details"
                      >
                        test entry
                      </a>
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                        data-testid="LockIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2m-6 9c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2m3.1-9H8.9V6c0-1.71 1.39-3.1 3.1-3.1s3.1 1.39 3.1 3.1z"
                        />
                      </svg>
                    </div>
                  </li>
                  <li
                    aria-hidden="true"
                    class="MuiBreadcrumbs-separator css-jhr02i-MuiBreadcrumbs-separator"
                  >
                    /
                  </li>
                  <li
                    class="MuiBreadcrumbs-li"
                  >
                    <p
                      class="MuiTypography-root MuiTypography-body1 css-1bgbmvw-MuiTypography-root"
                    >
                      変更履歴
                    </p>
                  </li>
                </ol>
              </nav>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root css-1cs9h4l"
        >
          <div
            class="MuiBox-root css-1519avi"
          >
            <div
              class="MuiBox-root css-9rtbay"
            >
              <h6
                class="MuiTypography-root MuiTypography-h6 css-2toz8d-MuiTypography-root"
                id="title"
              >
                test entry
              </h6>
              <h6
                class="MuiTypography-root MuiTypography-subtitle1 css-1i3tsf3-MuiTypography-root"
                id="description"
              >
                変更履歴
              </h6>
              <div
                class="MuiBox-root css-hoapfw"
              >
                <div
                  class="MuiBox-root css-kgitkj"
                >
                  <button
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-53g0n7-MuiButtonBase-root-MuiIconButton-root"
                    id=":r0:"
                    tabindex="0"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                      data-testid="AppsIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M4 8h4V4H4zm6 12h4v-4h-4zm-6 0h4v-4H4zm0-6h4v-4H4zm6 0h4v-4h-4zm6-10v4h4V4zm-6 4h4V4h-4zm6 6h4v-4h-4zm0 6h4v-4h-4z"
                      />
                    </svg>
                  </button>
                </div>
              </div>
            </div>
            <hr
              class="MuiDivider-root MuiDivider-fullWidth MuiDivider-flexItem css-lhgpb-MuiDivider-root"
            />
          </div>
        </div>
        <div
          class="MuiContainer-root MuiContainer-maxWidthLg css-5c1adp-MuiContainer-root"
        >
          <table
            class="MuiTable-root css-1xwxv7r-MuiTable-root"
            id="table_history_list"
          >
            <thead
              class="MuiTableHead-root css-1a7iywq-MuiTableHead-root"
            >
              <tr
                class="MuiTableRow-root MuiTableRow-head css-1wtsgsb-MuiTableRow-root"
              >
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-ljvtpv-MuiTableCell-root"
                  scope="col"
                  width="140px"
                >
                  項目
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-ljvtpv-MuiTableCell-root"
                  scope="col"
                  width="300px"
                >
                  変更前
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-ljvtpv-MuiTableCell-root"
                  scope="col"
                  width="300px"
                >
                  変更後
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-ljvtpv-MuiTableCell-root"
                  scope="col"
                  width="80px"
                >
                  実行日時
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-ljvtpv-MuiTableCell-root"
                  scope="col"
                  width="100px"
                >
                  実行者
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-ljvtpv-MuiTableCell-root"
                  scope="col"
                  width="40px"
                >
                  復旧
                </th>
              </tr>
            </thead>
            <tbody
              class="MuiTableBody-root css-gmh7jj-MuiTableBody-root"
            />
          </table>
          <div
            class="MuiBox-root css-5hgm98"
            id="pagination"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
            >
              0 - 0 / 0 件
            </p>
            <div
              class="MuiStack-root css-1sazv7p-MuiStack-root"
            >
              <nav
                aria-label="pagination navigation"
                class="MuiPagination-root MuiPagination-text css-1xra8g6-MuiPagination-root"
              >
                <ul
                  class="MuiPagination-ul css-1c5o7vy-MuiPagination-ul"
                >
                  <li>
                    <button
                      aria-label="Go to previous page"
                      class="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-colorPrimary MuiPaginationItem-textPrimary Mui-disabled MuiPaginationItem-previousNext css-1bug3cd-MuiButtonBase-root-MuiPaginationItem-root"
                      disabled=""
                      tabindex="-1"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-8uzxat-MuiSvgIcon-root-MuiPaginationItem-icon"
                        data-testid="NavigateBeforeIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                        />
                      </svg>
                    </button>
                  </li>
                  <li>
                    <button
                      aria-label="Go to next page"
                      class="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-colorPrimary MuiPaginationItem-textPrimary Mui-disabled MuiPaginationItem-previousNext css-1bug3cd-MuiButtonBase-root-MuiPaginationItem-root"
                      disabled=""
                      tabindex="-1"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-8uzxat-MuiSvgIcon-root-MuiPaginationItem-icon"
                        data-testid="NavigateNextIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                        />
                      </svg>
                    </button>
                  </li>
                </ul>
              </nav>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container MuiBox-root css-0"
    >
      <div
        class="MuiBox-root css-1baykhj"
      >
        <div
          class="MuiBox-root css-t0pdf1"
        >
          <div
            class="MuiBox-root css-jqffe8"
          >
            <nav
              aria-label="breadcrumb"
              class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-5cvx94-MuiTypography-root-MuiBreadcrumbs-root"
            >
              <ol
                class="MuiBreadcrumbs-ol css-1uwp4ue-MuiBreadcrumbs-ol"
              >
                <li
                  class="MuiBreadcrumbs-li"
                >
                  <a
<<<<<<< HEAD
                    class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
=======
                    class="MuiTypography-root MuiTypography-body1 css-z7oskh-MuiTypography-root"
>>>>>>> master
                    data-discover="true"
                    href="/ui/"
                  >
                    Top
                  </a>
                </li>
                <li
                  aria-hidden="true"
                  class="MuiBreadcrumbs-separator css-jhr02i-MuiBreadcrumbs-separator"
                >
                  /
                </li>
                <li
                  class="MuiBreadcrumbs-li"
                >
                  <a
<<<<<<< HEAD
                    class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
=======
                    class="MuiTypography-root MuiTypography-body1 css-z7oskh-MuiTypography-root"
>>>>>>> master
                    data-discover="true"
                    href="/ui/entities"
                  >
                    モデル一覧
                  </a>
                </li>
                <li
                  aria-hidden="true"
                  class="MuiBreadcrumbs-separator css-jhr02i-MuiBreadcrumbs-separator"
                >
                  /
                </li>
                <li
                  class="MuiBreadcrumbs-li"
                >
                  <div
                    class="MuiBox-root css-162tvoi"
                  >
                    <a
<<<<<<< HEAD
                      class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
=======
                      class="MuiTypography-root MuiTypography-body1 css-z7oskh-MuiTypography-root"
>>>>>>> master
                      data-discover="true"
                      href="/ui/entities/2/entries"
                    >
                      test entity
                    </a>
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                      data-testid="LockIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2m-6 9c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2m3.1-9H8.9V6c0-1.71 1.39-3.1 3.1-3.1s3.1 1.39 3.1 3.1z"
                      />
                    </svg>
                  </div>
                </li>
                <li
                  aria-hidden="true"
                  class="MuiBreadcrumbs-separator css-jhr02i-MuiBreadcrumbs-separator"
                >
                  /
                </li>
                <li
                  class="MuiBreadcrumbs-li"
                >
                  <div
                    class="MuiBox-root css-162tvoi"
                  >
                    <a
<<<<<<< HEAD
                      class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
=======
                      class="MuiTypography-root MuiTypography-body1 css-z7oskh-MuiTypography-root"
>>>>>>> master
                      data-discover="true"
                      href="/ui/entities/2/entries/1/details"
                    >
                      test entry
                    </a>
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                      data-testid="LockIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2m-6 9c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2m3.1-9H8.9V6c0-1.71 1.39-3.1 3.1-3.1s3.1 1.39 3.1 3.1z"
                      />
                    </svg>
                  </div>
                </li>
                <li
                  aria-hidden="true"
                  class="MuiBreadcrumbs-separator css-jhr02i-MuiBreadcrumbs-separator"
                >
                  /
                </li>
                <li
                  class="MuiBreadcrumbs-li"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-1bgbmvw-MuiTypography-root"
                  >
                    変更履歴
                  </p>
                </li>
              </ol>
            </nav>
          </div>
        </div>
      </div>
      <div
        class="MuiBox-root css-1cs9h4l"
      >
        <div
          class="MuiBox-root css-1519avi"
        >
          <div
            class="MuiBox-root css-9rtbay"
          >
            <h6
              class="MuiTypography-root MuiTypography-h6 css-2toz8d-MuiTypography-root"
              id="title"
            >
              test entry
            </h6>
            <h6
              class="MuiTypography-root MuiTypography-subtitle1 css-1i3tsf3-MuiTypography-root"
              id="description"
            >
              変更履歴
            </h6>
            <div
              class="MuiBox-root css-hoapfw"
            >
              <div
                class="MuiBox-root css-kgitkj"
              >
                <button
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-53g0n7-MuiButtonBase-root-MuiIconButton-root"
                  id=":r0:"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                    data-testid="AppsIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M4 8h4V4H4zm6 12h4v-4h-4zm-6 0h4v-4H4zm0-6h4v-4H4zm6 0h4v-4h-4zm6-10v4h4V4zm-6 4h4V4h-4zm6 6h4v-4h-4zm0 6h4v-4h-4z"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </div>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth MuiDivider-flexItem css-lhgpb-MuiDivider-root"
          />
        </div>
      </div>
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg css-5c1adp-MuiContainer-root"
      >
        <table
          class="MuiTable-root css-1xwxv7r-MuiTable-root"
          id="table_history_list"
        >
          <thead
            class="MuiTableHead-root css-1a7iywq-MuiTableHead-root"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-head css-1wtsgsb-MuiTableRow-root"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-ljvtpv-MuiTableCell-root"
                scope="col"
                width="140px"
              >
                項目
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-ljvtpv-MuiTableCell-root"
                scope="col"
                width="300px"
              >
                変更前
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-ljvtpv-MuiTableCell-root"
                scope="col"
                width="300px"
              >
                変更後
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-ljvtpv-MuiTableCell-root"
                scope="col"
                width="80px"
              >
                実行日時
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-ljvtpv-MuiTableCell-root"
                scope="col"
                width="100px"
              >
                実行者
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-ljvtpv-MuiTableCell-root"
                scope="col"
                width="40px"
              >
                復旧
              </th>
            </tr>
          </thead>
          <tbody
            class="MuiTableBody-root css-gmh7jj-MuiTableBody-root"
          />
        </table>
        <div
          class="MuiBox-root css-5hgm98"
          id="pagination"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
          >
            0 - 0 / 0 件
          </p>
          <div
            class="MuiStack-root css-1sazv7p-MuiStack-root"
          >
            <nav
              aria-label="pagination navigation"
              class="MuiPagination-root MuiPagination-text css-1xra8g6-MuiPagination-root"
            >
              <ul
                class="MuiPagination-ul css-1c5o7vy-MuiPagination-ul"
              >
                <li>
                  <button
                    aria-label="Go to previous page"
                    class="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-colorPrimary MuiPaginationItem-textPrimary Mui-disabled MuiPaginationItem-previousNext css-1bug3cd-MuiButtonBase-root-MuiPaginationItem-root"
                    disabled=""
                    tabindex="-1"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-8uzxat-MuiSvgIcon-root-MuiPaginationItem-icon"
                      data-testid="NavigateBeforeIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                      />
                    </svg>
                  </button>
                </li>
                <li>
                  <button
                    aria-label="Go to next page"
                    class="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-colorPrimary MuiPaginationItem-textPrimary Mui-disabled MuiPaginationItem-previousNext css-1bug3cd-MuiButtonBase-root-MuiPaginationItem-root"
                    disabled=""
                    tabindex="-1"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-8uzxat-MuiSvgIcon-root-MuiPaginationItem-icon"
                      data-testid="NavigateNextIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                      />
                    </svg>
                  </button>
                </li>
              </ul>
            </nav>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
